app/assets/javascripts/build_tags.js

```diff
@@ -0,0 +1,46 @@
+// Build chosen field for routing tag ids select on update batch action form
+
+$(document).on('mass_update_modal_dialog:after_open', function (event, form) {
+    var tagsChosen = $(form).find("select[name='routing_tag_ids']").prop({
+        'name': 'routing_tag_ids[]', 
+        'class': 'chosen', 
+        'id': 'routing_tag_ids',
+        'value': '', // reset default value
+        'multiple': true
+    });
+
+    if (tagsChosen.length === 0) {
+        return;
+    }
+
+    var hidden = $('<input>').prop({
+        'type': 'hidden',
+        'id': 'routing_tag_ids',
+        'name': 'routing_tag_ids',
+        'value': '',
+        'disabled': true
+    }).appendTo(form);
+
+    tagsChosen.chosen({no_results_text: "No results matched", width: '240px', search_contains: true, allow_single_deselect: true});
+
+    tagsChosen.change({hidden: hidden}, function(event){
+        if ($(event.target).val() != ''){
+            event.data.hidden.prop('disabled', true);
+        }else{
+            event.data.hidden.prop('disabled', false);
+        }
+    });
+});
+
+$(document).on('chosen:updated', function(event){
+    if ($(event.target).prop('id') != 'routing_tag_ids') { return; }
+    
+    var tagChosen = $('select#routing_tag_ids.chosen');
+    var disabled = tagChosen.prop('disabled')
+
+    if (disabled) { 
+        $("[name='routing_tag_ids']").prop('disabled', true); 
+    }else if(!disabled && tagChosen.val() == ''){
+        $("[name='routing_tag_ids']").prop('disabled', false); 
+    }
+});
```

Comment:
1. why did you add this handler on document instead of add it in upper function?
2. it looks like uglier duplicate of above code inside `tagsChosen.change({hidden: hidden}, function(event){`

Reply:
> 1. why did you add this handler on document instead of add it in upper function?
> 2. it looks like uglier duplicate of above code inside `tagsChosen.change({hidden: hidden}, function(event){`

This is not same event. This event triggered when user changed status of anyone checkbox.
When checkbox near tagChosen is not checked - hidden field must be disabled.
When checkbox is checked and tagChosen is not have any tags - hidden field must be enabled.